lib/mayaUsd/ufe/trf/UsdTransform3dMayaXformStack.cpp

```diff
@@ -374,12 +374,95 @@ MAYAUSD_VERIFY_CLASS_SETUP(UsdUfe::UsdTransform3dBase, UsdTransform3dMayaXformSt
 UsdTransform3dMayaXformStack::UsdTransform3dMayaXformStack(const UsdUfe::UsdSceneItem::Ptr& item)
     : UsdUfe::UsdTransform3dBase(item)
     , _xformable(prim())
+    , _needPivotConversion(isPivotConversionNeeded())
 {
     if (!TF_VERIFY(_xformable)) {
         throw std::runtime_error("Invalid scene item for transform stack");
     }
 }
 
+bool UsdTransform3dMayaXformStack::isPivotConversionNeeded() const
+{
+    // Note: USD and Maya use different pivots: USD has a single pivot that is used
+    //       for both translation and scale, while Maya has separate ones. When working
+    //       in this Maya transform stack mode, the USD pivot affects the position of
+    //       the manipulators, so we need to convert it to a Maya-style pivot.
+    //       Otherwise, prim with USD-style pivot won't work with the "center pivot"
+    //       command. They would also not work well with the universal manipulator.
+    TfToken pivotName
+        = UsdGeomXformOp::GetOpName(UsdGeomXformOp::TypeTranslate, UsdGeomTokens->pivot);
+    auto pivotAttr = prim().GetAttribute(pivotName);
+    if (!pivotAttr)
+        return false;
+
+    if (!pivotAttr.HasAuthoredValue())
+        return false;
+
+    Ufe::Vector3d pivot = getVector3d<GfVec3f>(pivotName);
+    if (isAlmostZero(pivot))
+        return false;
+
+    return true;
+}
+
+void UsdTransform3dMayaXformStack::convertToMayaPivotIfNeeded()
+{
+    if (!_needPivotConversion)
+        return;
+
+    // Note: must reset flag immediately because we call functions that would trigger
+    //       conversion again, resulting in infinite recursion.
+    _needPivotConversion = false;
+
+    // Extract and clear the USD common pivot. The exiesting pivot can be authored
+    // with any precision, so we need to convert it if needed.
+    GfVec3f commonPivotValue;
+    {
+        TfToken pivotName
+            = UsdGeomXformOp::GetOpName(UsdGeomXformOp::TypeTranslate, UsdGeomTokens->pivot);
+        UsdAttribute pivotAttr = prim().GetAttribute(pivotName);
+
+        VtValue currentValue;
+        if (!pivotAttr.Get(&currentValue, getTime(path())))
+            return;
+
+        if (currentValue.IsHolding<GfVec3f>()) {
+            commonPivotValue = currentValue.UncheckedGet<GfVec3f>();
+        } else if (currentValue.IsHolding<GfVec3d>()) {
+            auto val = currentValue.UncheckedGet<GfVec3d>();
+            commonPivotValue.Set(val[0], val[1], val[2]);
+        } else if (currentValue.IsHolding<GfVec3h>()) {
+            auto val = currentValue.UncheckedGet<GfVec3h>();
+            commonPivotValue.Set(val[0], val[1], val[2]);
+        } else {
+            commonPivotValue.Set(0, 0, 0);
+        }
+        pivotAttr.Set(GfVec3f(0, 0, 0));
+    }
+
+    // Adjust possibly existing Maya rotate pivot by the common pivot.
+    {
+        Ufe::Vector3d currentPivotValue = rotatePivot();
+        auto          cmd = rotatePivotCmd(
```

Comment:
If this is not an undoable command, you can just use rotePivt() which will automatically execute